### ISSUE TITLE
Newer API does not support "type"

### ISSUE THREAD

**Issue by Rocco83** [evgeni/katello-attach-subscription#17]
As by
https://github.com/RedHatSatellite/katello-attach-subscription/blob/e4c69df26b7831d3eaea523553c7e5826da4e1f5/katello-attach-subscription#L204

Newer API seems to not have the "type" filled.
- which is the root cause?
- how can we identify in the newer API the kind of server (Hypervisor, Physical, Virtual)
- should we block the yaml to run if type is set? -> reason why, we expect that what is written in the yaml is respected

**Comment by mossicrue** [comment]
With Hosts and Systems API unification, type field is currently missing for Satellite >= 6.2

Currently there is an open BugZilla for fix this problem, [BZ1635861](https://bugzilla.redhat.com/show_bug.cgi?id=1635861), that is linked to [Bug 11914](https://projects.theforeman.org/issues/11914) of Foreman

Also, in the BZ above is present a [possibile workaround](https://bugzilla.redhat.com/show_bug.cgi?id=1635861#c2) that can't fix at 100% the problem because assign the host type as hypervisor checking if name match `'*virt-who*'` regexp.

Another possible workaround (that can't fix the problem at 100% too) is to check the combination of this `facts` fields: `facts.virt`, `facts.virt::host_type` and `facts.virt::is_guest` values.

Depending on the servers type, this values are setted in this way:

Physical Server:
```
facts.virt::host_type : "Not Applicable"
facts.virt::is_guest  : false
facts.virt            : null
```

Guest:
```
facts.virt::host_type : "vmware"/"rhev, kvm"
facts.virt::is_guest  : true
facts.virt            : null
```

Hypervisor:
```
facts.virt::host_type : not present
facts.virt::is_guest  : not present
facts.virt            : not present
```

**Comment by evgeni** [comment]
fixed in 3821d6a4c03271b989b9bae9d745becab137aecb
